torch/csrc/utils/tensor_new.cpp

```diff
@@ -241,6 +246,11 @@ Tensor internal_new_from_data(
     if (copy_variables) {
       var = var.detach();
     }
+#ifdef BUILD_NAMEDTENSOR
+    if (names) {
```

Comment:
is it intentional that you test names for this propagate call but not for the others?

Reply:
Yes. This is case where we're passing a tensor to the `torch.tensor` constructor. It should behave differently if we pass in `names=None` (those are ignored) as opposed to `names=<something>`.

For example,
```
x: Tensor[N, C]
(1) torch.tensor(x): Tensor[N, C]. names=None implicitly.
(2) torch.tensor(x, names=None): Tensor[N, C]
torch.tensor(x, names=[N, D]): Error!
```
This is because we can't distinguish between cases (1) and (2).

Reply:
I can imagine that if we have a `x=tensor([1,2,3], names=['foo'])` and we try to call `y=torch.tensor(x)`, we would want to make sure `names` is preserved in `y`. Is that what you're trying to handle here?

What's the case that should error? I assume we should allow a user to change the names of a tensor in copy construction?

Reply:
I can see from the examples in tests that we don't allow changing names in construction. Should we / why don't we?

Reply:
We don’t allow it because there’s no use case for it and it is better to be conservative. If there is a use case for it, then there is a workaround and it is easy to add once someone asks for it.